enterprise/frontend/src/metabase-enterprise/upload_management/UploadManagementTable.tsx

```diff
@@ -0,0 +1,170 @@
+import type { Table } from "metabase-types/api";
+
+import { DeleteConfirmModal } from "./DeleteConfirmModal";
+import { getDateDisplay } from "./utils";
+
+const columns = [
+  { key: "checkbox", name: "" },
+  { key: "display_name", name: t`Table name` },
+  { key: "created_at", name: t`Created at` },
+  { key: "schema", name: t`Schema` },
+  { key: "actions", name: "" },
+];
+
+export function UploadManagementTable() {
+  const [selectedItems, setSelectedItems] = useState<Table[]>([]);
+  const [deleteTableRequest] = useDeleteUploadTableMutation();
+  const [showDeleteConfirmModal, setShowDeleteConfirmModal] = useState(false);
+
+  const { data: uploadTables, error, isLoading } = useListUploadTablesQuery();
+
+  const deleteTable = useCallback(
+    async (table: Table) => {
+      await deleteTableRequest({ tableId: table.id });
+    },
+    [deleteTableRequest],
+  );
+
+  const renderRow = useCallback(
+    (row: Table) => (
+      <UploadTableRow
+        item={row}
+        isSelected={selectedItems.some(
+          selectedTable => selectedTable.id === row.id,
+        )}
+        onSelect={newItem =>
+          setSelectedItems(prevItems => [...prevItems, newItem])
+        }
+        onDeselect={newItem =>
+          setSelectedItems(prevItems =>
+            prevItems.filter(i => i.id !== newItem.id),
+          )
+        }
+        onTrash={item => {
+          setSelectedItems([item]);
+          setShowDeleteConfirmModal(true);
+        }}
+      />
+    ),
+    [selectedItems],
+  );
+
+  if (isLoading || error) {
+    return <LoadingAndErrorWrapper loading={isLoading} error={error} />;
+  }
+
+  if (!uploadTables?.length) {
+    return null;
+  }
+
+  return (
+    <Box p="md" mt="lg">
+      <DeleteConfirmModal
+        opened={showDeleteConfirmModal}
+        tables={selectedItems}
+        onConfirm={async () => {
+          await Promise.all(selectedItems.map(deleteTable));
+          setShowDeleteConfirmModal(false);
+          setSelectedItems([]);
+        }}
+        onClose={() => setShowDeleteConfirmModal(false)}
+      />
+      {selectedItems.length > 0 && (
+        <BulkActionsPopover
+          opened={selectedItems.length > 0}
+          message={ngettext(
+            msgid`Selected ${selectedItems.length} table`,
+            `Selected ${selectedItems.length} tables`,
+            selectedItems.length,
+          )}
+        >
+          <BulkActionButton onClick={() => setShowDeleteConfirmModal(true)}>
+            {t`Delete`}
+          </BulkActionButton>
+        </BulkActionsPopover>
+      )}
+      <SettingHeader
+        id="upload-tables-list"
+        setting={{
+          display_name: t`Manage Uploads`,
+        }}
+      />
+      <Text fw="bold" color="text-medium">
+        {t`Uploaded Tables`}
+      </Text>
+      <StyledTable
+        data-testid="upload-tables-table"
+        columns={columns}
+        rows={uploadTables}
+        rowRenderer={row => renderRow(row as Table)}
```

Comment:
I have no idea why Typescript didn't like this, but was fine with the same structure in unit tests 🥴

Reply:
does seem to error for me locally...

Reply:
oh i figured out how to fix this in an upstream PR. it was an issue with emotion killing my generic types.